### PR TITLE
cats-effect: Add configF variants accepting Config

### PR DESCRIPTION
**Pull request by kubukoz**
Solves #388

### CODE REVIEW

**Comment by kubukoz** [comment]
btw. I had problems loading the `catseffect` module into IntelliJ, is there a way to actually have that working?

**Comment by kubukoz** [comment]
~Aw, I forgot to add scaladocs. Give me a sec~ Added

**Comment by kubukoz** [comment]
@leifwickland can we proceed with this one? :)

File: modules/cats-effect/src/test/scala/pureconfig/module/catseffect/CatsEffectSuite.scala [hunk 1]
```diff
@@ -47,6 +48,22 @@ class CatsEffectSuite extends BaseSuite {
     thrown.failures.head shouldBe a[ConvertFailure]
   }
 
+  it should "run successfully from a Typesafe Config object" in {
```
Comment (ruippeixotog) [inline modules/cats-effect/src/test/scala/pureconfig/module/catseffect/CatsEffectSuite.scala:51]: Can you add an example where the reader fails, both in `loadConfigF(Config)` and in `loadConfigF(Config, String)`?

**Review by ruippeixotog — COMMENTED** [review]
LGTM overall!

File: modules/cats-effect/src/test/scala/pureconfig/module/catseffect/CatsEffectSuite.scala (hunk 1 above)
Comment (kubukoz) [reply modules/cats-effect/src/test/scala/pureconfig/module/catseffect/CatsEffectSuite.scala:51]: Sure!

**Comment by kubukoz** [comment]
Added tests for invalid configs.

**Review by ruippeixotog — APPROVED** [review]
LGTM! Thanks, @kubukoz!

**Comment by kubukoz** [comment]
np, any estimates for when this could be merged/released?

**Comment by ruippeixotog** [comment]
I have just merged this. It will be available in the next pureconfig version. We still don't have a release date for it, but I can ping you here or in #388 once we release it.

**Comment by kubukoz** [comment]
I'll just wait until sbt-updates tells me there's a new one :P thanks

**Pull request merged**